src/main/kotlin/cacophony/semantic/names/NameResolver.kt

```diff
@@ -157,6 +157,16 @@ fun resolveNames(root: AST, diagnostics: Diagnostics): NameResolutionResult {
 
             is FunctionDeclaration -> {
                 symbolsTable.define(node.identifier, node)
+
+                node.arguments
+                    .groupBy { it.identifier }
+                    .filter { it.value.size > 1 }
+                    .values
+                    .flatten()
+                    .forEach { argNode ->
+                        diagnostics.report(NRDiagnostics.DuplicatedFunctionArgument(argNode.identifier), argNode.range)
+                    }
+
                 // Open new block to make arguments visible in function body, but not after
                 // the whole function declaration.
                 symbolsTable.open()
```

Comment:
I'm not sure if we should continue analyzing function when there's an error with the arguments 🤔

Reply:
Changed it to be fatal, we can refactor this after implementing structs and seeing how we handle them